File: Patchr/AirTableView.m
//
//  AirTableView.m
//  Patchr
//
//  Created by Jay Massena on 5/8/15.
//  Copyright (c) 2015 3meters. All rights reserved.
//

#import "AirTableView.h"

// From: http://stackoverflow.com/questions/19256996/uibutton-not-showing-highlight-on-tap-in-ios7/26049216#26049216

@implementation AirTableView

- (id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    
    if (self) {
        [self delayContentTouches];
    }
    return self;
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self delayContentTouches];
    }
    return self;
}

-(void)delayContentTouches {
    // iterate over all the UITableView's subviews
    for (id view in self.subviews) {
        
        // looking for a UITableViewWrapperView
        if ([NSStringFromClass([view class]) isEqualToString:@"UITableViewWrapperView"]) {
            
            // this test is necessary for safety and because a "UITableViewWrapperView" is NOT a UIScrollView in iOS7
            if([view isKindOfClass:[UIScrollView class]]) {
                
                // turn OFF delaysContentTouches in the hidden subview
                UIScrollView *scroll = (UIScrollView *) view;
                scroll.delaysContentTouches = NO;
            }
            break;
        }
    }
}

- (void)layoutSubviews {
    [super layoutSubviews];
    NSArray *sortedIndexPaths = [[self indexPathsForVisibleRows] sortedArrayUsingSelector:@selector(compare:)];
    for (NSIndexPath *path in sortedIndexPaths) {
        UITableViewCell *cell = [self cellForRowAtIndexPath:path];
        [self bringSubviewToFront:cell];
    }
}

-(void)scrollRectToVisible:(CGRect)rect animated:(BOOL)animated {
    // UIScrollView responds strangely when a textfield becomes first responder
    // http://stackoverflow.com/a/12640831/2247399
    return;
}

-(void)scrollToNearestSelectedRowAtScrollPosition:(UITableViewScrollPosition)scrollPosition animated:(BOOL)animated {
    return;
}

-(void)scrollToRowAtIndexPath:(NSIndexPath *)indexPath atScrollPosition:(UITableViewScrollPosition)scrollPosition animated:(BOOL)animated {
    return;
}

@end

File: Patchr/AirTableView.swift
//
//  AirTableView.swift
//  Patchr
//
//  Created by Jay Massena on 10/20/15.
//  Copyright © 2015 3meters. All rights reserved.
//

import UIKit

/*
 * Check this to investigate touch highlighting for buttons in scroll views.
 * From: http://stackoverflow.com/questions/19256996/uibutton-not-showing-highlight-on-tap-in-ios7/26049216#26049216
 */

class AirTableView: UITableView {
	
	required init(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)!
		initialize()
	}
	
	override init(frame: CGRect, style: UITableViewStyle) {
		super.init(frame: frame, style: style)
		initialize()
	}
	
	func initialize() {
		self.delaysContentTouches = false
	}
	
	override func scrollToNearestSelectedRow(at scrollPosition: UITableViewScrollPosition, animated: Bool) {
		return
	}
    
//    override func scrollToRow(at indexPath: IndexPath, at scrollPosition: UITableViewScrollPosition, animated: Bool) {
//		return
//	}
	
	override func touchesShouldCancel(in view: UIView) -> Bool {
		return true
	}
}
